### PR TITLE
fix: memory alloc fixes

### PR DESCRIPTION
**Pull request by m3taphysics**
## What does this PR change?

- Removes some simple allocations in SDK7

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ public class ECSScenesUiSystem : IDisposable
         private readonly BaseList<IParcelScene> loadedScenes;
         private readonly BaseVariable<bool> loadingHudVisibleVariable;
 
+        private static HashSet<IParcelScene> scenesToSort = new HashSet<IParcelScene>();
```
Comment (menduz) [inline unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: Please add comments why this change was required to prevent it from rolling back. Comment-less changes are fragile

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/PointerInputSystem/ECSPointerInputSystem.cs [hunk 2]
```diff
@@ -86,7 +85,7 @@ private static void Update(State state)
                 if (curState[i] != prevState[i])
                 {
                     PointerEventType pointerEventType = curState[i] ? PointerEventType.PetDown : PointerEventType.PetUp;
-                    InputAction inputAction = (InputAction)inputActionEnum[i];
+                    InputAction inputAction = (InputAction)_inputActionEnums[i];
```
Comment (pbosio) [inline unity-renderer/Assets/DCLPlugins/ECS7/Systems/PointerInputSystem/ECSPointerInputSystem.cs:88]: I think this could be just replaced by `inputAction inputAction = (InputAction)i;` and avoid the `Enum.GetValues` completely
@leanmendoza could you confirm?

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs (hunk 1 above)
Comment (pbosio) [inline unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: If we want to remove the allocation of the hashset, I wouldn't add that field as static.
I think it would probably be better to set it as part of the class state and modify `ApplyParenting` to receive it as argument

Comment (m3taphysics) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: Because it allocates memory at runtime causing memory fragmentation, increased memory usage, garbage collection stalls and hiccups

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/PointerInputSystem/ECSPointerInputSystem.cs (hunk 2 above)
Comment (m3taphysics) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/PointerInputSystem/ECSPointerInputSystem.cs:88]: This should work I will wait for @leanmendoza though

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs (hunk 1 above)
Comment (m3taphysics) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: What comment are you expecting?

Comment (menduz) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: Assuming there is a serial killer on the team, you need to prevent them to say "ugh, why is this static? I'll move it to a class property, it is not compatible with our standards" and then they introduce a regression in the allocations and want to come after you

Then something like `// This is a static property because ...`

Comment (pbosio) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: I wouldn't make that property as static, just part of the class instance state should be fine,
plus we avoid any "static property controversy" :)

Comment (m3taphysics) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: Agree! No idea why I didn't do it this way originally, good spot

Comment (m3taphysics) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs:21]: I agree with @pbosio it shouldn't be a static anyway!

**Review by mikhail-dcl — APPROVED** [review]

**Review by pbosio — APPROVED** [review]
